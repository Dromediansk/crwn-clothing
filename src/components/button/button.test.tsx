import { render, screen } from "@testing-library/react";
import Button from "./button.component";

it("should render default button", () => {
  render(<Button>Testing button</Button>);

  const button = screen.getByTestId("custom-button");

  expect(button).toMatchSnapshot();
});

it("should show spinner when loading", () => {
  render(<Button isLoading>Testing button</Button>);

  const button = screen.getByTestId("custom-button");
  const buttonSpinner = screen.getByTestId("button-spinner");

  expect(button).toBeDisabled();
  expect(button).toHaveTextContent("");
  expect(buttonSpinner).toBeInTheDocument();
});
