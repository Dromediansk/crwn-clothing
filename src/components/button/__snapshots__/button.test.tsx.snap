// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render default button 1`] = `
<button
  class="sc-dkrFOg fgcPVe"
  data-testid="custom-button"
>
  Testing button
</button>
`;
